include/glow/Backends/LayoutConverter.h

```diff
@@ -25,6 +25,9 @@ namespace glow {
 /// convolution nodes using NCHW.
 template <class NCHWConvNode>
 Node *convertConvToNCHWConv(ConvolutionNode *CN, Function *F) {
+  assert(CN->getGroup() == 1 &&
```

Comment:
This header is backend independent and meant to be shared, so the assert shouldn't go here. I think the other asserts you added are sufficient.

Reply:
Well, this is a place where we lose group parameter. It was present in initial node, but not in resulting node. So I think it's a good idea to have an assertion. I changed the text not to mention OpenCL.

Reply:
Ah ok got it. Perhaps we should support the group parameter in this function then, since other backends/users of this function might support it.

E.g. we could add group to the `OCLConvolutionNode`, and assert it's 1 when constructing the input to the OCL conv kernel.

Or we could pass another parameter to `convertConvToNCHWConv()` to signify if the group parameter should be included in the constructor for the new node. Then in `OCLBackend::transformPostLowering()` we could put this assertion that `group == 1`, and pass false for this parameter.

Reply:
@jfix71 : it makes sense to support Group Convolution both for NCHW layout and OpenCL, and any other layout where it is possible. The purpose of this PR is to add assert ASAP while support is not yet there :) It's much better than seeing wrong results as of now.